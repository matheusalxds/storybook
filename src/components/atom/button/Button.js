import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import css from './Button.module.scss';

const Button = ({ primary, success, danger, children }) => {
  const classes = classNames({
    [css['c-btn']]: true,
    [css['c-btn--primary']]: primary,
    [css['c-btn--danger']]: danger,
    [css['c-btn--success']]: success
  });
  return <button className={classes}>{children}</button>;
};

Button.propTypes = {
  primary: PropTypes.bool,
  danger: PropTypes.bool,
  success: PropTypes.bool,
  children: PropTypes.string.isRequired
};
Button.defaultProps = {
  primary: false,
  danger: false,
  success: false
};

export default Button;
