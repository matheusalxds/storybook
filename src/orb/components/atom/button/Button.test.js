import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from './Button';

describe('Button', () => {
  it('should match with the snapshot', () => {
    const children = 'Button';
    const wrapper = shallow(<Button>{children}</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
