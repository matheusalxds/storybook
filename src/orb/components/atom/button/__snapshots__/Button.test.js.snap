// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match with the snapshot 1`] = `
<button
  className="c-btn"
>
  Button
</button>
`;
